### PR TITLE
feat(ai-chat-log): Adding docs for new component

### PR DESCRIPTION
**Pull request by krisantrobus**
Added new docs page for `ai-chat-log`

### CODE REVIEW

File: .changeset/pretty-melons-punch.md [hunk 1]
```diff
@@ -3,4 +3,4 @@
 "@twilio-paste/core": patch
 ---
 
-[Button] Add border radius 20 to size="reset" buttons which can be overridden by passing a border radius token to variant="reset" and size="reset" buttons
\ No newline at end of file
+[Button] Add border radius 20 to size="reset" buttons which can be overridden by passing a border radius token to variant="reset" and size="reset" buttons
```
Comment (krisantrobus) [inline .changeset/pretty-melons-punch.md:6]: not sure why this is picked up as a change

File: packages/paste-core/components/ai-chat-log/package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.0",
   "category": "data display",
   "status": "production",
-  "description": "Ai chat log.",
+  "description": "An AI Chat Log is a collection of AI Chat components for displaying conversations between a human and an AI bot.",
```
Comment (nkrantz) [inline packages/paste-core/components/ai-chat-log/package.json:6]: niceeee

File: packages/paste-website/src/component-examples/AIChatLogExamples.ts [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+          <AIChatMessageActionCard>
+            Is this helpful?
+            <Button variant="reset" size="reset">
+              <ThumbsUpIcon decorative={false} title="like result" />
+            </Button>
+            <Button variant="reset" size="reset">
+              <ThumbsDownIcon decorative={false} title="dislike result" />
+            </Button>
+          </AIChatMessageActionCard>
+          <AIChatMessageActionCard>
```
Comment (nkrantz) [inline packages/paste-website/src/component-examples/AIChatLogExamples.ts:54]: suggestion: I think I added some aria labels to the rewrite button and maybe also the thumbs up/down buttons in the stories - can you copy whatever I did there for the action card sections in these examples?

File: packages/paste-website/src/pages/components/ai-chat-log/index.mdx [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+<LivePreview
+  scope={{
+    AIChatLog,
+    AIChatMessage,
+    AIChatMessageAuthor,
+    AIChatMessageBody,
+  }}
+  language="jsx"
+>
+  {`<AIChatLog>
```
Comment (nkrantz) [inline packages/paste-website/src/pages/components/ai-chat-log/index.mdx:79]: question (non-blocking): just curious, nothing wrong with it, but why are you writing this example out here and importing the others?

File: packages/paste-website/src/pages/components/chat-log/index.mdx [hunk 5]
```diff
@@ -292,7 +292,7 @@ This example combines all the separate features displayed previously into one ex
 
 The `useChatLogger` hook provides a hook based approach to managing chat state. It is best used with the `<ChatLogger />` component.
 
-`useChatLogger` returns 3 things:
+`useChatLogger` returns 4 things:
```
Comment (nkrantz) [inline packages/paste-website/src/pages/components/chat-log/index.mdx:295]: 🤣 good catch

File: packages/paste-website/src/pages/components/ai-chat-log/index.mdx [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+    AIChatMessage,
+    AIChatMessageAuthor,
+    AIChatMessageBody,
+    Button,
+    ButtonGroup,
+  }}
+  language="jsx"
+  noInline
+>
+  {botWithBodyActions}
```
Comment (nkrantz) [inline packages/paste-website/src/pages/components/ai-chat-log/index.mdx:230]: issue(blocking): these buttons need some padding, can you also go down to just 3 buttons so we're not showing off the wrapping bug until we fix it 😸 

image

**Review by nkrantz — COMMENTED** [review]
So good!! Can you add a little section to explain the props for ChatMessageAuthor to change the content of the avatar?

File: packages/paste-website/src/component-examples/AIChatLogExamples.ts (hunk 3 above)
Comment (krisantrobus) [reply packages/paste-website/src/component-examples/AIChatLogExamples.ts:54]: Great catch. I will address this in the next commit

File: packages/paste-website/src/pages/components/ai-chat-log/index.mdx (hunk 4 above)
Comment (krisantrobus) [reply packages/paste-website/src/pages/components/ai-chat-log/index.mdx:79]: The first LivePreview in the chat-log index page did the same. Seeing a trend for simpler components while looking through other doc pages not being put in examples. 

I open to changing this if we want to start a best practice on it

File: packages/paste-website/src/pages/components/ai-chat-log/index.mdx (hunk 6 above)
Comment (krisantrobus) [reply packages/paste-website/src/pages/components/ai-chat-log/index.mdx:230]: Absolutely!

File: packages/paste-website/src/pages/components/ai-chat-log/index.mdx (hunk 4 above)
Comment (nkrantz) [reply packages/paste-website/src/pages/components/ai-chat-log/index.mdx:79]: Makes sense, no need to change :)

**Review by roxanagomez — COMMENTED** [review]
issue (blocking): move the error codes in the example to the next line. 
Screenshot 2024-06-27 at 8 44 29 AM

File: packages/paste-website/src/pages/components/ai-chat-log/index.mdx [hunk 7]
```diff
@@ -0,0 +1,363 @@
...
+    </AIChatMessageBody>
+  </AIChatMessage>
+</AIChatLog>`}
+</LivePreview>
+
+## Guidelines
+
+### About AI Chat Log
+
+An AI Chat Log is a way to display conversations between a user and AI. If you are looking for a chat between 2 or more humans please refer to [ChatLog](/components/chat-log).
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/ai-chat-log/index.mdx:102]:
```suggestion
An AI Chat Log is a way to display conversations between a user and AI. If you are looking for a chat between 2 or more humans, please refer to [ChatLog](/components/chat-log).
```

File: packages/paste-website/src/pages/components/ai-chat-log/index.mdx [hunk 8]
```diff
@@ -0,0 +1,363 @@
...
+
+The only other accessibility requirement is providing the `AIChatMessageActionCard` a descriptive label via the `aria-label` React prop.
+
+The AIChatLog component has `role="log"` which means that any new messages added to it are announced by assistive technology.
+
+## Examples
+
+### Basic Message
+
+A message must include the author and body. Any message text from a user or a bot must be contained within the `AIChatMessageBody` component. Due to lengthy AI responses, the chat layout is now top-down.
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/ai-chat-log/index.mdx:126]:
```suggestion
A message must include the author and body. Any message text from a user or a bot must be contained within the `AIChatMessageBody` component. Due to lengthy AI responses, the chat layout is top-down.
```

File: packages/paste-website/src/pages/components/ai-chat-log/index.mdx [hunk 9]
```diff
@@ -0,0 +1,363 @@
...
+  }}
+  language="jsx"
+  noInline
+>
+  {aiChatLoggerExample}
+</LivePreview>
+
+## Composition Notes
+
+Keep any generated responses from the AI contained in the `AIChatMessageBody` component. Each chat message should only have one `AIChatMessageBody` component.import { AvatarExample } from "../../../component-examples/SkeletonLoaderExamples";
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/ai-chat-log/index.mdx:363]:
```suggestion
Keep any generated responses from the AI contained in the `AIChatMessageBody` component. Each chat message should only have one `AIChatMessageBody` component.
```

**Review by nkrantz — APPROVED** [review]

**Review by serifluous — APPROVED** [review]

**Pull request merged**